### PR TITLE
Several small documentation updates

### PR DESCRIPTION
**Pull request by paulromano**
This PR makes a few small updates in the documentation, mostly having to do with installation.

### CODE REVIEW

File: docs/source/io_formats/statepoint.rst [hunk 1]
```diff
@@ -4,7 +4,7 @@
 State Point File Format
 =======================
 
-The current version of the statepoint file format is 17.0.
+The current version of the statepoint file format is 18.1.
```
Comment (eepeterson) [inline docs/source/io_formats/statepoint.rst:7]: why did we go to 18.1 rather than 18.0?

File: docs/source/devguide/workflow.rst [hunk 2]
```diff
@@ -116,7 +116,7 @@ pip_. From the root directory of the OpenMC repository, run:
 
 .. code-block:: sh
 
-    pip install -e .[test]
+    python -m pip install -e .[test]
```
Comment (eepeterson) [inline docs/source/devguide/workflow.rst:119]: Out of curiosity when did this become the best practice and what exactly does it help with? Does it make sure that OpenMC is installed in the distribution where `python` is available and avoid the chance that the `pip` being used is from another distro?

**Review by eepeterson — APPROVED** [review]
Just a couple questions, but no requested changes!

File: docs/source/devguide/workflow.rst (hunk 2 above)
Comment (paulromano) [reply docs/source/devguide/workflow.rst:119]: Long discussion of this [here](https://snarky.ca/why-you-should-use-python-m-pip/)

**Comment by eepeterson** [comment]
Awesome thanks @paulromano

**Pull request merged**